### PR TITLE
Added instructions for delay needed after SDDC creation for site recovery.

### PR DESCRIPTION
**Pull request by saloneerege**

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
-This is an example that demonstrates SDDC management actions like creating, updating and deleting an existing SDDC.
+This is an example that demonstrates the following workflow:
 
-To run the example:
+1. SDDC management actions like creating, updating and deleting an existing SDDC.
+2. Public IP management actions like creation and deletion.
+3. Site recovery management actions like activation and deactivation.
+4. SRM node management actions like creation and deletion.
+
+For this workflow, a 10 minute delay must be added after SDDC is created and before site recovery can be activated.
```
Comment (rchadalawada) [inline examples/README.md]: For which workflow? 1, 2, 3 or 4? Each of the workflow steps seem to be an individual workflow by itself.

File: examples/README.md [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+    customer_subnet_ids  = [data.vmc_customer_subnets.my_subnets.ids[0]]
+    connected_account_id = data.vmc_connected_accounts.my_accounts.id
+  }
+  timeouts {
+    create = "300m"
+    update = "300m"
+    delete = "180m"
+  }
+   provisioner "local-exec" {
+   command = "sleep 600"
```
Comment (rchadalawada) [inline examples/README.md]: indentation

File: vmc/resource_vmc_site_recovery.go [hunk 3]
```diff
@@ -72,6 +72,10 @@ func resourceSiteRecovery() *schema.Resource {
 
 func resourceSiteRecoveryCreate(d *schema.ResourceData, m interface{}) error {
 
+	err := (m.(*ConnectorWrapper)).authenticate()
```
Comment (rchadalawada) [inline vmc/resource_vmc_site_recovery.go:75]: does this check if the current token is valid or not and if not, get a new token OR it always gets a new token?

File: examples/README.md (hunk 1 above)
Comment (saloneerege) [reply examples/README.md]: The main.tf in the examples folder is for all the steps 1,2,3 and 4 combined. That is why we need to add the provisioner local exec which adds the 10 minute delay which is mentioned on line 10. 
I have created a separate PR for each individual workflows too.

File: vmc/resource_vmc_site_recovery.go (hunk 3 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go:75]: authenticate() creates a new access token with a validity of 30 minutes. There is no other API but the POST API /am/api/auth/api-tokens/authorize to refresh the access token currently under this authentication mechanism. The access token can be refreshed only in the OAuth authentication mechanism using the refresh_token grant type.

File: examples/README.md (hunk 2 above)
Comment (saloneerege) [reply examples/README.md]: Done

File: examples/README.md [hunk 4]
```diff
@@ -1,13 +1,59 @@
 # Example: Provision an SDDC
```
Comment (shylasrinivas) [inline examples/README.md]: with all the workflows listed below, does this heading still apply?

File: examples/README.md (hunk 1 above)
Comment (rchadalawada) [reply examples/README.md]: My point is that we don’t do all the actions you described in each of the steps. The workflow is more like create sddc, create public up, enable site recovery and finally SRM node creation.

File: examples/README.md (hunk 4 above)
Comment (saloneerege) [reply examples/README.md]: Good catch. I will change it

File: examples/README.md (hunk 1 above)
Comment (saloneerege) [reply examples/README.md]: Agreed but terraform apply would perform creation on the resources but with a simple terraform destroy you can delete the resources too. Or in the same main.tf if someone were to update some field on any of the resources that would do an update action. Hence, I have listed all of them.

File: vmc/resource_vmc_site_recovery.go (hunk 3 above)
Comment (shylasrinivas) [reply vmc/resource_vmc_site_recovery.go:75]: I think we should rename this func https://github.com/terraform-providers/terraform-provider-vmc/blob/master/vmc/provider.go#L27. Quite misleading.

File: vmc/resource_vmc_site_recovery.go [hunk 5]
```diff
@@ -72,6 +72,10 @@ func resourceSiteRecovery() *schema.Resource {
...
+	if err != nil {
+		return fmt.Errorf("Error authenticating in CSP: %s", err)
```
Comment (shylasrinivas) [inline vmc/resource_vmc_site_recovery.go]: Is this error displayed on console during apply? If yes, does CSP make sense to users?

File: vmc/resource_vmc_site_recovery.go (hunk 3 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go:75]: Yes I agree.

File: vmc/resource_vmc_site_recovery.go (hunk 5 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go]: Yes this error does get displayed during apply. This is a CSP API and although I did not work on this error message earlier I used the same one from https://github.com/terraform-providers/terraform-provider-vmc/blob/master/vmc/resource_vmc_sddc.go#L277 .  I can change it everywhere to Cloud Server Provider to be consistent in the code.

File: examples/README.md [hunk 6]
```diff
@@ -1,13 +1,54 @@
-# Example: Provision an SDDC
+# Example: Creation of SDDC, public IP, site recovery and SRM node.
 
-This is an example that demonstrates SDDC management actions like creating, updating and deleting an existing SDDC.
+This is an example that demonstrates the creation of VMC resources like SDDC, public IP, site recovery and SRM node.
+
+For site recovery activation,a 10 minute delay must be added after SDDC is created and before site recovery can be activated.
```
Comment (shylasrinivas) [inline examples/README.md]: typo space after comma

File: examples/README.md (hunk 4 above)
Comment (saloneerege) [reply examples/README.md]: Done

File: vmc/resource_vmc_site_recovery.go (hunk 5 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go]: Done

File: examples/README.md (hunk 6 above)
Comment (saloneerege) [reply examples/README.md]: Done.

File: examples/README.md (hunk 1 above)
Comment (saloneerege) [reply examples/README.md]: Done

File: examples/README.md [hunk 7]
```diff
@@ -1,13 +1,54 @@
...
+      }   
+    }
+
+```
 
 To run the example:
 
 * Generate an API token using [VMware Cloud on AWS console] (https://vmc.vmware.com/console/)
 
 * Update the variables required parameters api_token and org_id in [variables.tf](https://github.com/terraform-providers/terraform-provider-vmc/blob/master/examples/variables.tf) with your infrastructure settings.
```
Comment (shylasrinivas) [inline examples/README.md]: can you remove either variables or parameters?

Comment (saloneerege) [reply examples/README.md]: Done

**Review by shylasrinivas — APPROVED** [review]

**Pull request merged**